sandbox/grist/dropdown_condition.py

```diff
@@ -26,24 +26,31 @@ def visit_Attribute(self, node):
 
 def perform_dropdown_condition_renames(useractions, renames):
   """
-  Given a dict of column renames of the form {(table_id, col_id): new_col_id}, applys updates
+  Given a dict of column renames of the form {(table_id, col_id): new_col_id}, applies updates
   to the affected dropdown condition formulas.
   """
   updates = []
 
-  for col in useractions._engine.docmodel.columns.all:
+  for col in useractions._docmodel.columns.all:
 
     patches = []
 
-    # Find all columns in the document that has dropdown conditions.
+    # Find all columns in the document that have dropdown conditions.
     try:
       widget_options = json.loads(col.widgetOptions)
       dc_formula = widget_options["dropdownCondition"]["text"]
     except (json.JSONDecodeError, KeyError):
       continue
 
     # Find out what table this column refers to and belongs to.
-    ref_table_id = col.type.lstrip("Ref:")
+    if col.type.startswith("Ref:") :
+      ref_table_id = col.type[4:] 
+    elif col.type.startswith("RefList:"):
+      ref_table_id = col.type[8:] 
+    else:
+      # Unexpected situation. A column with dropdown condition should be of type reference
```

Comment:
Choice and Choice List columns support dropdown conditions too. They work a little different: `choice` is just a string (and doesn't have any attributes that need renaming). But we should still rename any instances of `$`/`rec` that need renaming.

Actually, it would be good to always rename `$`/`rec`, regardless of column type. Dropdown conditions stay in widget options, even when the current column type can't make use of them. You can see this when you cycle between column types that support and don't support dropdown conditions. For columns that aren't Choice or Reference [Lists], it's not possible to know whether `choice` was a string or a record, but we can still rename the `$`/`rec` instances, if needed.

Reply:
Thanks for the info! I made some changes and added a test case for the situation you mentioned. Hopefully that addresses your comment.